test/metabase/lib/test_util/generators/filters.cljc

```diff
@@ -0,0 +1,278 @@
+    (#{:type/BigInteger :type/Integer} effective-type) (gen-int)
+    (lib.types.isa/category? column)                   (gen-category)
+    (lib.types.isa/latitude? column)                   (gen-latitude)
+    (lib.types.isa/longitude? column)                  (gen-longitude)
+    (lib.types.isa/numeric? column)                    (cond-> (gen-int)
+                                                         (< (rand) 0.5) (+ (rand)))
+    (lib.types.isa/string-or-string-like? column)      (gen-string)
+    (lib.types.isa/time? column)                       (gen-time)
+    (lib.types.isa/date-without-time? column)          (gen-date)
+    (isa? effective-type :type/DateTime)               (gen-datetime)
+    :else (throw (ex-info " !!! Not sure what values to generate for column" {:effective-type effective-type
+                                                                              :column         column}))))
+
+;; Filter clauses ================================================================================
+(def ^:private ^:dynamic *filterable-columns* nil)
+
+(defmulti ^:private gen-filter-clause
+  (fn [_column operator]
+    (:short operator)))
+
+;; Binary operators like :<
+(doseq [[op f] [[:<  lib/<]
+                [:<= lib/<=]
+                [:>  lib/>]
+                [:>= lib/>=]]]
+  (defmethod gen-filter-clause op [column _op]
+    (f column (rand-column-value column))))
+
+;; Multi-value operators like := and :starts-with
+(doseq [[op f] [[:=                lib/=]
+                [:!=               lib/!=]
+                [:starts-with      lib/starts-with]
+                [:ends-with        lib/ends-with]
+                [:contains         lib/contains]
+                [:does-not-contain lib/does-not-contain]]]
+  (defmethod gen-filter-clause op [column _op]
+    (apply f column (repeatedly (inc (rand-int 4))
+                                #(rand-column-value column)))))
+
+(defmethod gen-filter-clause :between [column _op]
+  (let [lo (rand-column-value column)
+        hi (rand-column-value column)]
+    ;; TODO: Maybe make the LHS arg be the smaller? Right now they're random.
+    ;; Probably they can be `sort`ed, but that might fail on weird types.
+    (lib/between column lo hi)))
+
+;; Unary operators like `:is-empty`.
+(doseq [[op f] [[:is-empty  lib/is-empty]
+                [:not-empty lib/not-empty]
+                [:is-null   lib/is-null]
+                [:not-null  lib/not-null]]]
+  (defmethod gen-filter-clause op [column _op]
+    (f column)))
+
+(defn- skipped-operator? [op]
+  (#{:inside} (:short op)))
+
+(defn- gen-filter:unit [column]
+  (gen.u/choose (if (lib.types.isa/date-without-time? column)
+                  [:day :week :month :quarter :year]
+                  [:minute :hour :day :week :month :quarter :year])))
+
+(defn- gen-filter:relative-current [column]
+  (lib/time-interval column :current (gen-filter:unit column)))
+
+(defn- gen-filter:relative-date [column]
```

Comment:
There are also is an offset option available, in this case we use `relative-time-interval`, see https://github.com/metabase/metabase/blob/4aa67506aa001805ef99b47239465b2e8f82dc5d/frontend/src/metabase-lib/filter.ts#L336

Reply:
If there is an offset then `include-current` is not supported